guardian/editions
# hide nav bar and slider on Android when reading article

## Summary

On Android phones with small screen space, the reading experience has been worse than it could be. This changeset make it so we hide the bars at the top when scrolling down and show them again when navigating left and right to other articles, or scrolling up.

That's a little rework of #799.

I also upgraded the mechanism that displays a thin line below the slider if the article gets scrolled down (both for iOS and Android). Now it only sends a single message when the status changes, instead of before where it was sending tons of messages.

I made it so there is very little re-renders on state changes throughout.

## Test Plan

On the emulator the animation is fairly slow, so I tested the final version on a real device.

1. Open app, then open an article
2. Scroll down as if reading the article. Observe the header disappear.
3. Continue scrolling down, then scroll up. Observe the header show up after a bit.
4. Scroll down again, notice it hides pretty fast.
5. Swipe right to the next article, notice it shows up again
6. Swipe left to the original article, notice it keeps showing.
7. Scroll down, notice it hides up again as expected.

Video of a slightly older version, on emulator: https://youtu.be/DJOZgP5NPzA
(unlike in the video, there is an animation up and down of the while bar)
